# Heart Disease Research Part I
In this project, we’ll investigate some data from a sample of patients who were evaluated for heart disease at the Cleveland Clinic Foundation. The data was downloaded from the UCI Machine Learning Repository and then cleaned for analysis. The principal investigators responsible for data collection were:

- Hungarian Institute of Cardiology. Budapest: Andras Janosi, M.D.
- University Hospital, Zurich, Switzerland: William Steinbrunn, M.D.
- University Hospital, Basel, Switzerland: Matthias Pfisterer, M.D.
- V.A. Medical Center, Long Beach and Cleveland Clinic Foundation: Robert Detrano, M.D., Ph.D.

In [1]:
# import libraries
import pandas as pd
import numpy as np
from scipy.stats import ttest_1samp
from scipy.stats import binom_test

In [2]:
# load data
heart = pd.read_csv('heart_disease.csv')

yes_hd = heart[heart.heart_disease == 'presence']
no_hd = heart[heart.heart_disease == 'absence']

## Cholesterol Analysis

The full dataset has been loaded for us as heart, then split into two subsets:

- **yes_hd**, which contains data for patients with heart disease
- **no_hd**, which contains data for patients without heart disease
  
For this project, we’ll investigate the following variables:

- **chol:** serum cholestorol in mg/dl
- **fbs:** An indicator for whether fasting blood sugar is greater than 120 mg/dl (1 = true; 0 = false)

To start, we’ll investigate cholesterol levels for patients with heart disease. We'll use the dataset yes_hd to save cholesterol levels for patients with heart disease as a variable named chol_hd.

In [3]:
chol_hd = yes_hd.chol

In general, total cholesterol over 240 mg/dl is considered “high” (and therefore unhealthy). We will calculate the mean cholesterol level for patients who were diagnosed with heart disease and print it out. Is it higher than 240 mg/dl?

In [4]:
print(np.mean(chol_hd))

251.4748201438849


Do people with heart disease have high cholesterol levels (greater than or equal to 240 mg/dl) on average? Let's work with a hyphotesis tool that we can use to test the following null and alternative hypotheses:

- **Null:** People with heart disease have an average cholesterol level equal to 240 mg/dl
- **Alternative:** People with heart disease have an average cholesterol level that is greater than 240 mg/dl
  
**Note:** Unfortunately, the scipy.stats function we’ve been using does not (at the time of writing) have an alternative parameter to change the alternative hypothesis for this test. Therefore, we’ll have to run a two-sided test. However, since we calculated earlier that the average cholesterol level for heart disease patients is greater than 240 mg/dl, we can calculate the p-value for the one-sided test indicated above simply by dividing the two-sided p-value in half.

In [5]:
tstat, pval = ttest_1samp(chol_hd, 240)
real_pval = pval/2
print(real_pval)

0.0035411033905155707


**Result:** p value of 0.0035, a significant value that indicates the alternative hypothesis is true. This suggests that heart disease patients have an average cholesterol level higher than 240 mg/dl

We'll repeat the previous steps in order to run the same hypothesis test, but for patients in the sample who were not diagnosed with heart disease. Do patients without heart disease have average cholesterol levels significantly above 240 mg/dl?

In [6]:
chol_no_hd = no_hd.chol
print(np.mean(chol_no_hd))

242.640243902439


In [7]:
tstat_no_hd, pval_no_hd = ttest_1samp(chol_no_hd, 240)
real_pval_no_hd = pval_no_hd
print(real_pval_no_hd)  

0.5279424046444101


**Results:** p value = 0.52 , no significance so the null hyphotesis is true.

## Fasting Blood Sugar Analysis 

Let’s now return to the full dataset (saved as heart). How many patients are there in this dataset? Let's save the number of patients as num_patients and print it out.

In [8]:
num_patients = len(heart)
print(num_patients)

303


Remember that the fbs column of this dataset indicates whether or not a patient’s fasting blood sugar was greater than 120 mg/dl (1 means that their fasting blood sugar was greater than 120 mg/dl; 0 means it was less than or equal to 120 mg/dl).

Let's calculate the number of patients with fasting blood sugar greater than 120. Save this number as num_highfbs_patients and print it out

In [9]:
num_highfbs_patients = np.sum(heart.fbs)
print(num_highfbs_patients)

45.0


Sometimes, part of an analysis will involve comparing a sample to known population values to see if the sample appears to be representative of the general population.

By some estimates, about 8% of the U.S. population had diabetes (diagnosed or undiagnosed) in 1988 when this data was collected. While there are multiple tests that contribute to a diabetes diagnosis, fasting blood sugar levels greater than 120 mg/dl can be indicative of diabetes (or at least, pre-diabetes). If this sample were representative of the population, approximately how many people would you expect to have diabetes? Let's calculate and print out this number.

Is this value similar to the number of patients with a resting blood sugar above 120 mg/dl — or different?

In [10]:
proportion_fbs = (num_highfbs_patients / num_patients) * 100
print(proportion_fbs)

14.85148514851485


**Results:** about 14.85 % of the population of this dataset has diabetes or pre-diabetes

Does this sample come from a population in which the rate of fbs > 120 mg/dl is equal to 8%? Let's try to choose one of these hyphotesis using statistic tools:

- **Null:** This sample was drawn from a population where 8% of people have fasting blood sugar > 120 mg/dl
- **Alternative:** This sample was drawn from a population where more than 8% of people have fasting blood sugar > 120 mg/dl

In [11]:
p_value_fbs = binom_test(45, 303, 0.08, alternative='greater')
print(p_value_fbs)

4.689471951448875e-05


**Results:** the p-value is significant so we choose the alternative hypothesis 